### ISSUE TITLE
Unhandled Exception!: TypeError: Cannot call method 'toLowerCase' of null

### ISSUE THREAD

**Issue by rltvty** [mscdex/node-imap#477]
Lately we have been getting many instances of this error at this line: https://github.com/mscdex/node-imap/blob/master/lib/Parser.js#L502

Unfortunately I do not have a test case yet to reproduce the problem.  I will attempt to get one soon.

**Comment by rltvty** [comment]
using the following test script (with username & password removed):

```
var Imap = require('imap'),
  inspect = require('util').inspect;

var imap = new Imap({
  user: 'removed',
  password: 'removed',
  host: 'imap.exmail.qq.com',
  port: 993,
  tls: true,
  tlsOptions: { rejectUnauthorized: false },
  debug: console.log
});

function openInbox(cb) {
  imap.openBox('INBOX', true, cb);
}

imap.once('ready', function() {
  openInbox(function(err, box) {
    if (err) throw err;
    var f = imap.fetch('13718', {
      struct: true
    });
    f.on('message', function(msg, seqno) {
      console.log('Message #%d', seqno);
      var prefix = '(#' + seqno + ') ';
      msg.on('body', function(stream, info) {
        var buffer = '';
        stream.on('data', function(chunk) {
          buffer += chunk.toString('utf8');
        });
        stream.once('end', function() {
          console.log(prefix + 'Parsed header: %s', inspect(Imap.parseHeader(buffer)));
        });
      });
      msg.once('attributes', function(attrs) {
        console.log(prefix + 'Attributes: %s', inspect(attrs, false, 8));
      });
      msg.once('end', function() {
        console.log(prefix + 'Finished');
      });
    });
    f.once('error', function(err) {
      console.log('Fetch error: ' + err);
    });
    f.once('end', function() {
      console.log('Done fetching all messages!');
      imap.end();
    });
  });
});

imap.once('error', function(err) {
  console.log('Connection error: ' + err.message);
});

imap.once('end', function() {
  console.log('Connection ended');
});

imap.connect();
```

Received the following log and IMAP trace and exception (with username & password removed):

```
[connection] Connected to host
<= '* OK [CAPABILITY IMAP4 IMAP4rev1 IDLE XAPPLEPUSHSERVICE ID UIDPLUS AUTH=LOGIN NAMESPACE] QQMail IMAP4Server ready'
=> 'A0 CAPABILITY'
<= '* CAPABILITY IMAP4 IMAP4rev1 IDLE XAPPLEPUSHSERVICE AUTH=LOGIN NAMESPACE CHILDREN ID UIDPLUS'
<= 'A0 OK CAPABILITY Completed'
=> 'A1 LOGIN "removed" "removed"'
<= 'A1 OK Success login ok'
=> 'A2 CAPABILITY'
<= '* CAPABILITY IMAP4 IMAP4rev1 IDLE XAPPLEPUSHSERVICE NAMESPACE CHILDREN ID UIDPLUS'
<= 'A2 OK CAPABILITY Completed'
=> 'A3 NAMESPACE'
<= '* NAMESPACE (("" "/")) NIL NIL'
<= 'A3 OK Success'
=> 'A4 LIST "" ""'
<= '* LIST (\\NoSelect) "/" "/"'
<= 'A4 OK LIST completed'
=> 'A5 EXAMINE "INBOX"'
<= '* 812 EXISTS'
<= '* 0 RECENT'
<= '* OK [UIDVALIDITY 1394260498] UID validity status'
<= '* OK [UIDNEXT 13750] Predicted next UID'
<= '* FLAGS (\\Answered \\Flagged \\Deleted \\Draft \\Seen)'
<= '* OK [PERMANENTFLAGS ()] No permanent flags permitted'
<= 'A5 OK [READ-ONLY] EXAMINE complete'
=> 'A6 UID FETCH 13718 (UID FLAGS INTERNALDATE BODYSTRUCTURE)'
<= '* 781 FETCH (UID 13718 FLAGS (\\Answered \\Seen) INTERNALDATE "10-Jun-2015 05:02:27 +0800" BODYSTRUCTURE (NIL NIL ("CHARSET" "GB2312") NIL NIL NIL 176 NIL NIL NIL))'

.../node_modules/imap/lib/Parser.js:502
        part = { type: cur[0].toLowerCase(), params: null };
                              ^
TypeError: Cannot call method 'toLowerCase' of null
    at parseBodyStructure (.../node_modules/imap/lib/Parser.js:502:31)
    at parseBodyStructure (.../node_modules/imap/lib/Parser.js:466:13)
    at Parser.parseFetch (.../node_modules/imap/lib/Parser.js:440:13)
    at Parser._resUntagged (.../node_modules/imap/lib/Parser.js:257:24)
    at Parser._parse (.../node_modules/imap/lib/Parser.js:137:16)
    at Parser._tryread (.../node_modules/imap/lib/Parser.js:82:15)
    at CleartextStream.Parser._cbReadable (.../node_modules/imap/lib/Parser.js:53:12)
    at CleartextStream.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:427:10)
    at _stream_readable.js:420:7
```

This is the only message I've found in this account that creates this exception.

**Comment by mscdex** [comment]
Fixed by #478
